Sources/MongoSwift/ChangeStream.swift

```diff
@@ -0,0 +1,165 @@
+import mongoc
+
+/// A token used for manually resuming a change stream. Pass this to the `resumeAfter` or `startAfter` fields of
+/// `ChangeStreamOptions` to resume or start a change stream where a previous one left off.
+/// - SeeAlso: https://docs.mongodb.com/manual/changeStreams/#resume-a-change-stream
+public struct ChangeStreamToken: Codable {
+    private let resumeToken: Document
+
+    public init(resumeToken: Document) {
+        self.resumeToken = resumeToken
+    }
+
+    public init(from decoder: Decoder) throws {
+      self.resumeToken = try Document(from: decoder)
+    }
+}
+
+/// A MongoDB ChangeStream.
+/// - SeeAlso: https://docs.mongodb.com/manual/changeStreams/
+public class ChangeStream<T: Codable>: Sequence, IteratorProtocol {
+  /// A `ChangeStreamToken` used for manually resuming a change stream.
+  public private(set) var resumeToken: ChangeStreamToken
+
+  /// A `MongoClient` stored to make sure the source client stays valid until the change stream is destroyed.
+  private var client: MongoClient?
```

Comment:
since change streams cannot be manually closed, I think we can make `client`, `session`, and `changeStream` all non-optional and immutable.

Reply:
Oops, that's right. This was changed with SPEC-1343. Updated.